src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -178,6 +184,67 @@ bool DoorLockServer::SetPrivacyModeButton(chip::EndpointId endpointId, bool isEn
     return SetAttribute(endpointId, Attributes::EnablePrivacyModeButton::Id, Attributes::EnablePrivacyModeButton::Set, isEnabled);
 }
 
+bool DoorLockServer::TrackWrongCodeEntry(chip::EndpointId endpointId)
+{
+    auto endpointContext = getContext(endpointId);
+    if (nullptr == endpointContext)
+    {
+        ChipLogError(Zcl, "Failed to get endpoint index for cluster [endpoint=%d]", endpointId);
+        return false;
+    }
+
+    uint8_t wrongCodeEntryLimit = 0xFF;
+    auto status                 = Attributes::WrongCodeEntryLimit::Get(endpointId, &wrongCodeEntryLimit);
+    if (EMBER_ZCL_STATUS_SUCCESS == status)
+    {
+        if (++endpointContext->wrongCodeEntryAttempts >= wrongCodeEntryLimit)
+        {
+            emberAfDoorLockClusterPrintln("Too many wrong code entry attempts, engaging lockout [endpoint=%d,wrongCodeAttempts=%d]",
+                                          endpointId, endpointContext->wrongCodeEntryAttempts);
+            engageLockout(endpointId);
+        }
+    }
+    else if (EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE != status)
+    {
+        ChipLogError(Zcl, "Failed to read Wrong Code Entry Limit attribute, status=0x%x", to_underlying(status));
+        return false;
+    }
+    return true;
+}
+
+bool DoorLockServer::engageLockout(chip::EndpointId endpointId)
```

Comment:
Where is this return value used?  Should it be used somewhere?

Reply:
Not right now but maybe in the future... it's more like a habit to put a return value everywhere and not using it...